ext/spl/spl_directory.c

```diff
@@ -1939,18 +1939,18 @@ static int spl_filesystem_file_read_line_ex(zval * this_ptr, spl_filesystem_obje
 
 	/* 1) use fgetcsv? 2) overloaded call the function, 3) do it directly */
 	if (SPL_HAS_FLAG(intern->flags, SPL_FILE_OBJECT_READ_CSV) || intern->u.file.func_getCurr->common.scope != spl_ce_SplFileObject) {
-		spl_filesystem_file_free_line(intern);
-
-		if (php_stream_eof(intern->u.file.stream)) {
-			if (!silent) {
-				zend_throw_exception_ex(spl_ce_RuntimeException, 0, "Cannot read from file %s", intern->file_name);
-			}
-			return FAILURE;
-		}
 		if (SPL_HAS_FLAG(intern->flags, SPL_FILE_OBJECT_READ_CSV)) {
```

Comment:
Move this outside the outer if, as it no longer shares code?

Reply:
Ah, right. Thank you!